Content.Client/SSDIndicator/SSDIndicatorSystem.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Content.Shared.CCVar;
+using Content.Shared.SSDIndicator;
+using Content.Shared.StatusIcon;
+using Content.Shared.StatusIcon.Components;
+using Robust.Shared.Configuration;
+using Robust.Shared.Prototypes;
+
+namespace Content.Client.SSDIndicator;
+
+/// <summary>
+///     Handles displaying SSD indicator as status icon
+/// </summary>
+public sealed class SSDIndicatorSystem : EntitySystem
+{
+    [Dependency] private readonly IPrototypeManager _prototype = default!;
+    [Dependency] private readonly IConfigurationManager _cfg = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SSDIndicatorComponent, GetStatusIconsEvent>(OnGetStatusIcon);
+    }
+
+    private void OnGetStatusIcon(EntityUid uid, SSDIndicatorComponent component, ref GetStatusIconsEvent args)
+    {
+        if (!component.IsSSD || !_cfg.GetCVar(CCVars.ICShowSSDIndicator))
+            return;
+
+        args.StatusIcons.Add(_prototype.Index<StatusIconPrototype>(component.Icon));
+    }
+}
```

Comment:
I think this should be a UI controller but checking with @DrSmugleaf  first.

Reply:
> at some point yeah that should be in a controller, right now thats fine being kept in a system since it doesnt use anything under user interface like controls and overlays are pretty well separated from the rest of the code